cli/cmd/agent_aws-install_ec2ssm.go

```diff
@@ -0,0 +1,350 @@
+
+    lacework agent aws-install ec2ssm --skip_iam_role_creation
+
+To provide a preexisting IAM role with the 'SSMManagedInstanceCore' policy
+
+    lacework agent aws-install ec2ssm --iam_role_name IAMRoleName
+
+To filter by one or more regions:
+
+    lacework agent aws-install ec2ssm --include_regions us-west-2,us-east-2
+
+To filter by instance tag:
+
+    lacework agent aws-install ec2ssm --tag TagName,TagValue
+
+To filter by instance tag key:
+
+    lacework agent aws-install ec2ssm --tag_key TagName
+
+To provide an agent access token of your choice, use the command 'lacework agent token list',
+select a token and pass it to the '--token' flag. This flag must be selected if the
+'--noninteractive' flag is set.
+
+    lacework agent aws-install ec2ssm --token <token>
+
+AWS credentials are read from the following environment variables:
+- AWS_ACCESS_KEY_ID
+- AWS_SECRET_ACCESS_KEY
+- AWS_SESSION_TOKEN (optional)
+- AWS_REGION`,
+	}
+)
+
+func init() {
+	// 'agent aws-install ec2ssm' flags
+	agentInstallAWSSSMCmd.Flags().StringVar(&agentCmdState.InstallTagKey,
+		"tag_key", "", "only install agents on infra with this tag key set",
+	)
+	agentInstallAWSSSMCmd.Flags().StringSliceVar(&agentCmdState.InstallTag,
+		"tag", []string{}, "only install agents on infra with this tag",
+	)
+	agentInstallAWSSSMCmd.Flags().StringSliceVarP(&agentCmdState.InstallIncludeRegions,
+		"include_regions", "r", []string{}, "list of regions to filter on",
+	)
+	agentInstallAWSSSMCmd.Flags().StringVar(&agentCmdState.InstallAgentToken,
+		"token", "", "agent access token",
+	)
+	agentInstallAWSSSMCmd.Flags().IntVarP(
+		&agentCmdState.InstallMaxParallelism,
+		"max_parallelism",
+		"n",
+		50,
+		"maximum number of workers executing AWS API calls, set if rate limits are lower or higher than normal",
+	)
+	agentInstallAWSSSMCmd.Flags().StringVar(
+		&agentCmdState.InstallBYORole,
+		"iam_role_name",
+		"",
+		"IAM role name (not ARN) with SSM policy, if not provided then an ephemeral role will be created",
+	)
+	agentInstallAWSSSMCmd.Flags().BoolVar(
+		&agentCmdState.InstallSkipCreatInfra,
+		"skip_iam_role_creation",
+		false,
+		"set this flag to skip creating an IAM role and instance profile and associating the instance profile. Assumes all instances are already setup for SSM",
+	)
+	agentInstallAWSSSMCmd.Flags().BoolVarP(
+		&agentCmdState.InstallDryRun,
+		"dry_run",
+		"d",
+		false,
+		"set this flag to print out the target instances and exit",
+	)
+}
+
+func installAWSSSM(_ *cobra.Command, _ []string) error {
+	token := agentCmdState.InstallAgentToken
+	if token == "" {
+		if cli.InteractiveMode() {
+			// user didn't provide an agent token
+			cli.Log.Debugw("agent token not provided, asking user to select one now")
+			var err error
+			token, err = selectAgentAccessToken()
+			if err != nil {
+				return err
+			}
+		} else {
+			return errors.New("user did not provide or interactively select an agent token")
+		}
```

Comment:
nit; reduce if statement logic and better error message to the end-user
```suggestion
		if !cli.InteractiveMode() {
			return errors.New("agent token not provided. Use '--token' when running in non interactive mode")
		}
		// user didn't provide an agent token
		cli.Log.Debug("agent token not provided, asking user to select one now")
		var err error
		token, err = selectAgentAccessToken()
		if err != nil {
			return err
		}
```

Reply:
Good idea, done in 17a10310cea74528271622b52e1303a6d298a00e